packages/flutter/lib/src/material/bottom_navigation.dart

```diff
@@ -0,0 +1,536 @@
+          );
+        }
+
+        bottomNavigation = new SizedBox(
+          width: _maxWidth,
+          child: new Row(
+            children: children
+          )
+        );
+
+        break;
+      case BottomNavigationType.shifting:
+        final List<Widget> children = <Widget>[];
+
+        _computeWeight();
+
+        for (int i = 0; i < config.actions.length; i += 1) {
+          children.add(
+            new Flexible(
+              flex: (_ratio(_animations[i]) * 1000.0).round(),
+              child: new InkResponse(
+                onTap: () {
+                  if (config.onAction != null)
+                    config.onAction(_animations[i], i);
+
+                  _pushCircle(i);
+
+                  setState(() {
+                    _controllers[currentIndex].reverse();
+                    currentIndex = i;
+                    _controllers[currentIndex].forward();
+                  });
+                },
+                child: new Stack(
+                  alignment: FractionalOffset.center,
+                  children: <Widget>[
+                    new Align(
+                      alignment: FractionalOffset.topCenter,
+                      child: new Container(
+                        margin: new EdgeInsets.only(
+                          top: new Tween<double>(
+                            begin: 18.0,
+                            end: 6.0
+                          ).evaluate(_animations[i])
+                        ),
+                        child: new IconTheme(
+                          data: new IconThemeData(
+                            color: Colors.white
+                          ),
+                          child: config.actions[i].icon
+                        )
+                      )
+                    ),
+                    new Align(
+                      alignment: FractionalOffset.bottomCenter,
+                      child: new Container(
+                        margin: const EdgeInsets.only(bottom: 10.0),
+                        child: new FadeTransition(
+                          opacity: _animations[i],
+                          child: new DefaultTextStyle(
+                            style: new TextStyle(
+                              fontSize: 14.0,
+                              color: Colors.white
+                            ),
+                            child: config.actions[i].label
+                          )
+                        )
+                      )
+                    )
+                  ]
+                )
+              )
+            )
+          );
+        }
+
+        bottomNavigation = new SizedBox(
+          width: _maxWidth,
+          child: new Row(
+            children: children
+          )
+        );
+
+        break;
+    }
+
+    return new Stack(
+      children: <Widget>[
+        new Positioned(
+          left: 0.0,
+          top: 0.0,
+          right: 0.0,
+          bottom: 0.0,
+          child: new Material(
```

Comment:
why is there a material here? Materials are pretty heavy-weight, but you don't seem to actually use this one for anything other than a background color and shadow? Why not put it outside the stack? I don't really follow.